docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -90,4 +97,135 @@ hs.refreshIndex("empIndex", "incremental")
+deleted files in their source dataset.
+
+###### How to enable
+
+The following configurations turn on Hybrid Scan append-only. You need to call `spark.enableHyperspace`
+after setting the configuration to load additional modules for Hybrid Scan.
+
+Scala:
+```scala
+import com.microsoft.hyperspace._
+
+spark.conf.set("spark.hyperspace.index.hybridscan.enabled", true)
+spark.conf.set("spark.hyperspace.index.hybridscan.delete.enabled", false) // false by default
+spark.enableHyperspace
+```
+
+Python:
+```python
+from hyperspace import Hyperspace
+
+spark.conf.set("spark.hyperspace.index.hybridscan.enabled", true)
+spark.conf.set("spark.hyperspace.index.hybridscan.delete.enabled", false) // false by default
+Hyperspace.enable(spark)
+```
+
+###### Example
+
+This is a simple example in Scala from [Quick-Start Guide](https://microsoft.github.io/hyperspace/docs/ug-quick-start-guide/).
+Of course, you can try this in Python accordingly.
+
+```scala
+// Setup source data.
+import org.apache.spark.sql._
+import spark.implicits._
+
+Seq((1, "name1"), (2, "name2")).toDF("id", "name").write.mode("overwrite").parquet("table")
+val df = spark.read.parquet("table")
+
+// Setup Hyperspace.
+import com.microsoft.hyperspace._
+val hs = new Hyperspace(spark)
+
+// Create an index.
+import com.microsoft.hyperspace.index._
+hs.createIndex(df, IndexConfig("index", indexedColumns = Seq("id"), includedColumns = Seq("name")))
+
+// Create a query and check if the index is applied or not.
+val query = df.filter(df("id") === 1).select("name")
+hs.explain(query, verbose = true)
+
+// Run query with the index.
+spark.enableHyperspace
+query.show
+```
+
+Now, the following example shows how Hybrid Scan works with appended files.
+
+```scala
+// Append new data to source dataset.
+Seq((3, "name3"), (4, "name4")).toDF("id", "name").write.mode("append").parquet("table")
+
+// Check if the index is applied for the dataset with appended files.
+val df = spark.read.parquet("table")
+val query = df.filter(df("id") === 1).select("name")
+hs.explain(query, verbose = true)
+
+// Turn on Hybrid Scan and check if the index is applied.
+spark.conf.set("spark.hyperspace.index.hybridscan.enabled", true)
+hs.explain(query, verbose = true)
+
+// Query execution with Hybrid Scan.
+spark.enableHyperspace
+query.show
+```
+
+#### Append and Delete dataset
+
+Now, we can consider handling deleted files. Basically, Hybrid Scan excludes indexed data from deleted source files
+by scanning all indexed rows and confirming if it is from deleted source files or not.
+In order to trace which source file each row is from, you need to enable linage column config before creating an index.
+Check the [configuration](https://microsoft.github.io/hyperspace/docs/ug-configuration/) page to see how lineage is enabled when creating an index.
+
+Because of this nature of processing deleted rows, Hybrid Scan with deleted files is more expensive than appended files.
```

Comment:
suggestion:
"Due to the way Hybrid Scan enforces deletes at the query time, supporting deletes is more expensive than appended files and it could cause performance regression if an index has a large number of deleted source data files."

Reply:
I revised a bit:
Due to the way Hybrid Scan enforces deletes at the query time, supporting deletes is more expensive than appended files. It could cause performance regression if an index has a number of deleted source data files, especially for an index which is less effective for a given query.

Reply:
if you prefer to drop "large" from "a large number of deleted source ...", you may wanna change it as:
"If an index has some deleted source data files".

My only concern about above rephrase is: "... an index which is less effective for a given query.".
Today, we dont have a way to measure how effective an index is for a query (we have not yet formally defined it either).
So, if a user asks: "How can I know if Index XX is less effective for query Q?" - We dont really have a clear answer for that.
WDYT @sezruby ? Should we drop the "less effective" part (or change it)?

Reply:
I revised a bit. I thought `it could cause performance regression if an index has a large number of deleted source data files.` seems like there will be always perf regression with many deleted files.